### PR TITLE
added separation for box and levers

### PR DESCRIPTION
**Pull request by Addehman**
won't be able to place the box on the same x-value of the lever, it will now glide away from it if they're too close.

### CODE REVIEW

File: Assets/Scripts/Player/PlayerHandDetection.cs [hunk 1]
```diff
@@ -57,15 +56,21 @@ public class PlayerHandDetection : MonoBehaviour {
...
-			if (distance < lowestDistance) {
+			if (distance > lowestDistance && interactable.tag == "Box") {
+				lowestDistance = distance;
+				closestItem = interactable;				
+			}
+
+			else if (distance < lowestDistance) {
 				lowestDistance = distance;
 				closestItem = interactable;
 			}
```
Comment (MrJackey) [inline Assets/Scripts/Player/PlayerHandDetection.cs]:
```suggestion
			if (distance < lowestDistance || interactable.CompareTag("Box")) {
				lowestDistance = distance;
				closestItem = interactable;
			}
```

File: Assets/Scripts/Player/PlayerHandDetection.cs [hunk 2]
```diff
@@ -57,15 +56,21 @@ public class PlayerHandDetection : MonoBehaviour {
 		foreach (Interactable interactable in interactableList) {
 			float distance = (transform.position - interactable.transform.position).sqrMagnitude;
 
-			if (distance < lowestDistance) {
+			if (distance > lowestDistance && interactable.tag == "Box") {
```
Comment (MrJackey) [inline Assets/Scripts/Player/PlayerHandDetection.cs]: Maybe add a serializedfield to select priority?

File: Assets/Scripts/Puzzle/Box.cs [hunk 3]
```diff
@@ -24,6 +24,17 @@ public class Box : MonoBehaviour {
...
+		if (other.CompareTag("Lever")) {
+			float amount = 0.1f * Time.deltaTime;
+			float pushAwayZone = 0.2f;
+			if (transform.position.x >= other.transform.position.x - pushAwayZone && transform.position.x <= other.transform.position.x)
+				transform.Translate(-amount, 0, 0);
+			else if (transform.position.x <= other.transform.position.x + pushAwayZone && transform.position.x > other.transform.position.x)
+				transform.Translate(amount, 0, 0);
+		}
+	}
+
```
Comment (MrJackey) [inline Assets/Scripts/Puzzle/Box.cs]: Do you really need this push if the box has priority over the lever?

**Review by MrJackey — CHANGES_REQUESTED** [review]
A few things, take a look

**Review by MrJackey — APPROVED** [review]
+1

File: Assets/Scripts/Player/PlayerHandDetection.cs [hunk 4]
```diff
@@ -57,15 +57,16 @@ public class PlayerHandDetection : MonoBehaviour {
 		foreach (Interactable interactable in interactableList) {
 			float distance = (transform.position - interactable.transform.position).sqrMagnitude;
 
-			if (distance < lowestDistance) {
+			if (distance < lowestDistance || interactable.CompareTag(priorityInteractableTag)) {
```
Comment (smeas) [inline Assets/Scripts/Player/PlayerHandDetection.cs]: Don't check priority if no tag is selected.
```suggestion
			if (distance < lowestDistance || (!string.IsNullOrEmpty(priorityInteractableTag) && interactable.CompareTag(priorityInteractableTag))) {
```

File: Assets/Scripts/Player/PlayerHandDetection.cs [hunk 5]
```diff
@@ -57,15 +57,16 @@ public class PlayerHandDetection : MonoBehaviour {
...
 				lowestDistance = distance;
-				closestItem = interactable;
+				closestItem = interactable;
```
Comment (smeas) [inline Assets/Scripts/Player/PlayerHandDetection.cs]:
```suggestion
				closestItem = interactable;
```

**Review by smeas — CHANGES_REQUESTED** [review]

File: Assets/Scripts/Player/PlayerHandDetection.cs (hunk 5 above)
Comment (Addehman) [reply Assets/Scripts/Player/PlayerHandDetection.cs]: I think you are pointing to the tabs after the 62 line? X)

**Review by smeas — APPROVED** [review]

**Pull request merged**